src/components/HelpScoutBeacon/HelpScoutBeacon.js

```diff
@@ -0,0 +1,465 @@
+import React, { useCallback, useEffect, useState } from 'react'
+import PropTypes from 'prop-types'
+import styled from 'styled-components'
+import { Transition, animated } from 'react-spring'
+import {
+  Button,
+  ButtonIcon,
+  IconClose,
+  SafeLink,
+  breakpoint,
+  springs,
+  theme,
+  useViewport,
+} from '@aragon/ui'
+import BeaconHeadScripts from './BeaconHeadScripts'
+import IconQuestion from './IconQuestion'
+import headerImg from './header.png'
+import { useClickOutside } from '../../hooks'
+import LoadingRing from '../LoadingRing'
+import { GU } from '../../utils'
+
+const HELPSCOUT_BEACON_KEY = 'helpscout-beacon'
+const CLOSED = Symbol('closed, user can open opt-in dialogue')
+const OPENED = Symbol('opened, user can opt-in or close')
+const OPENING = Symbol('opening')
+const CLOSING = Symbol('closing')
+const ROUND_BUTTON_HEIGHT = 40
+
+const Beacon = React.memo(function Beacon() {
+  const [beaconReady, setBeaconReady] = useState(false)
+  const [openOnReady, setOpenOnReady] = useState(false)
+  const [optedIn, setOptedIn] = useState(
+    localStorage.getItem(HELPSCOUT_BEACON_KEY) === '1'
+  )
+
+  const handleOptIn = () => {
+    localStorage.setItem(HELPSCOUT_BEACON_KEY, '1')
+    setOptedIn(true)
+    setOpenOnReady(true)
+  }
+  const handleBeaconReady = useCallback(() => {
+    if (openOnReady && window.Beacon) {
+      window.Beacon('open')
+      window.Beacon('once', 'open', () => {
+        setBeaconReady(true)
+        setOpenOnReady(false)
+      })
+    } else {
+      setBeaconReady(true)
+    }
+  }, [openOnReady])
+
+  return (
+    <div
+      css={`
+        position: absolute;
+        bottom: ${2 * GU}px;
+        right: ${2 * GU}px;
+        z-index: 4;
+
+        ${breakpoint(
+          'medium',
+          `
+            z-index: 10000;
+            bottom: ${3 * GU}px;
+            right: ${3 * GU}px;
+          `
+        )}
+      `}
+    >
+      <BeaconHeadScripts optedIn={optedIn} onReady={handleBeaconReady} />
+      <HelpOptIn
+        beaconReady={beaconReady}
+        onOptIn={handleOptIn}
+        optedIn={optedIn}
+      />
+    </div>
+  )
+})
+
+const HelpOptIn = React.memo(function HelpOptIn({
+  beaconReady,
+  onOptIn,
+  optedIn,
+}) {
+  const { above } = useViewport()
+  const [mode, setMode] = useState(CLOSED)
+
+  const handleClose = React.useCallback(() => setMode(CLOSED), [])
+  const handleToggle = useCallback(() => {
+    if (mode !== OPENING && mode !== CLOSING) {
+      setMode(mode === CLOSED ? OPENING : CLOSING)
+    }
+    if (beaconReady && window.Beacon) {
+      window.Beacon('toggle')
+    }
+  }, [beaconReady, mode])
+  const handleToggleEnd = useCallback(() => {
+    setMode(mode === OPENING ? OPENED : CLOSED)
+  }, [mode])
+  const { ref } = useClickOutside(() => {
+    if ((mode === OPENED || mode === OPENING) && above('medium')) {
+      handleToggle()
+    }
+  }, [mode, handleToggle, above('medium')])
```

Comment:
I think here, we should separate out the callback from the `useClickOutside`. Right now, the `cb` would change on every render, re-running `useClickOutside`'s `useEffect()`.

We should instead wrap this function in a `useCallback()` with the necessary dependencies, and then pass that to `useClickOutside()`:

```js
const handleClickOutside = useCallback(() => {
  if ((mode === OPENED || mode === OPENING) && above('medium')) {
    handleToggle()
  }
}, [mode, handleToggle, above('medium')])
const { ref } = useClickOutside(handleClickOutside)
```